src/utils/mutate.ts

```diff
@@ -100,7 +100,7 @@ export const internalMutate = async <Data>(
   if (populateCache) {
     if (!error) {
       // Transform the result into data.
-      if (isFunction(populateCache)) {
+      if (isFunction(populateCache) && originalData) {
```

Comment:
Without this check, there was this error:

```
Argument of type 'Data | undefined' is not assignable to parameter of type 'Data'.
```

This is because `populateCache` expects the arg to be `Data`,

https://github.com/vercel/swr/blob/dfcba4d63026bee6b2994a5c23ab6fdc2a327263/src/types.ts#L147

but `originalData` can be `undefined`, as we're getting it from the cache:

https://github.com/vercel/swr/blob/dfcba4d63026bee6b2994a5c23ab6fdc2a327263/src/utils/mutate.ts#L55

However, not sure if this will break something.

Reply:
@shuding suggested to change the argument type of `populateCache` to accept `undefined` instead, closing this comment.